### ISSUE TITLE
Spike for Contests Farcaster Integration

### ISSUE THREAD

**Issue by sachben91** [hicommonwealth/commonwealth#7478]

**Comment by ForestMars** [comment]
Super quick spike to answer 3 feasibility & estimation questions about. 30-90 mins timebox on protocol side. @Rotorsoft for visibility.

**Comment by ForestMars** [comment]
TBH would like to do this on tomorrow (halving day! but not for that reason obvi.)

**Comment by ianrowan** [comment]
1. Login/create Account with frame:
    - As we discussed this makes mores sense as a frame/action to create a common user via frame as 'signing in' isnt necessary with farcaster as a client
    - This is pretty easy to do and would basically require the user to provide a user name in the frame and then we receive a newUser request that first checks if the farcaster connected address is active(revert if true) and then creates a User/profile/Address with the Address and input. Note we don't need session key's here as farcaster itself technically manages the session
    - Low effort platform team. low effort PENG for frame
2. Yes this would require a bridge similar to the discord bridge from Farcaster -> Common. Mapping the resultant casts/replies is arbitrary but would need to be well scoped product wise. Effort involved here would be medium, ie maybe a sprint for platform team to build.
3. Upvotes would be built in with the above actions and effort included in medium estimate

Going to pull through a slack message with a general estimate of all Faracaster things we'd want to do for contests as well

"
Basically we'll need:
PENG - figure out FC frame tagging for HEAD of contest pages(requires design)
Platform - Farcaster -> common relay and model mapping + routes for frames & actions
Protocol - align contract helpers for contest with the product requirements of our farcaster support + any thing custom required for this that is off model from the contest helper implementations we have
"

CC @sachben91 @dillchen
